package/src/components/CatalogGridItem/v1/CatalogGridItem.js

```diff
@@ -113,20 +113,38 @@ class CatalogGridItem extends Component {
   };
 
   componentDidMount() {
+    this._mounted = true;
+
+    this.setImageFit();
+  }
+
+  componentDidUpdate() {
+    this.setImageFit();
+  }
+
+  componentWillUnmount() {
+    this._mounted = false;
+  }
+
+  setImageFit = () => {
     // Use cover fit if image is landcape, contain if portrait
     if (typeof Image !== "undefined") {
       const { large } = this.primaryImage.URLs;
       const largeImage = new Image();
       largeImage.src = large;
       largeImage.onload = () => {
+        if (this._mounted === false) {
+          return;
+        }
+
         const { width, height } = largeImage;
         if (height > width) {
           // Image is portrait
           this.setState({ fit: "contain" });
         }
```

Comment:
To be safe, maybe add

```
else {
  this.setState({ fit: "cover" });
}
```

Because otherwise state might still be at "contain" since you are now calling it multiple times, right? (The same problem this fixes, but in reverse.)